superset/utils/core.py

```diff
@@ -1292,6 +1292,19 @@ def get_user_email() -> str | None:
         return None
 
 
+def get_user_roles() -> list[str] | None:
```

Comment:
Not sure why we need this method in `core.py`. (?) Seems you could simply use this `[role.name for role in g.user.roles]` in `superset/jinja_context.py`